galsim/utilities.py

```diff
@@ -330,34 +349,137 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
         # Nothing to do
         return x,f
 
+    total_integ = np.trapz(abs(f), x)
+    thresh = total_integ * rel_err
+
+    if not preserve_range:
+        errleft = 0.5 * (abs(f[1])+abs(f[0])) * (x[1]-x[0])
+        errright = 0.5 * (abs(f[-1])+abs(f[-2])) * (x[-1]-x[-2])
+        accumulated_err = 0.0
+        # arbitrarily allocate half of error budget to trimming the endpoints.
+        # while at least one endpoint can be trimmed:
```

Comment:
I noticed in the test_thin output that I posted, this algorithm actually trims less when not preserving the range than when it does.  I think this "arbitrary" decision is the reason for that.  I think what is happening is that half your error budget is going into trimming a few points off each end, leaving not much more than half the budget left for the main section.  So it seems that this part of the algorithm may actually be counter-productive.

I think maybe adopting the prescription used by the fixed version of the old code would be better, where the error budget for trimming the ends is taken to be thresh times the fraction of the x range that would be removed.  So if you are trimming off 20% of the x range, you get to use 20% of the total error budget.